Fix #11

michaelkourlas opened the pull request.

giacomodrago commented on minijson_reader.hpp:
```diff
@@ -636,6 +638,11 @@ void read_quoted_string(Context& context, bool skip_opening_quote = false)
                 {
                     const uint16_t code_unit = parse_utf16_escape_sequence(utf16_seq);
 
+                    if (code_unit == 0)
```
This isn't quite right I think. If you are parsing a _surrogate pair_ (a very fringe edge case of UTF-16 that I am sure is handled very badly pretty much everywhere, but here I am at least trying to get almost right) and you find 0x0000 as the low code unit, that's not a null character, it is just invalid. I guess the right thing to do is to perform the check wherever it is that I come up with the canonical 32 bit representation of the Unicode character and if it is zero there, then it is most definitely a valid null character. If you can be bothered to figure that out that's great, otherwise I'll see what I can do. Thanks.

michaelkourlas commented:
I've updated the PR to return INVALID_UTF16_CHARACTER when \u0000 is a low surrogate, as before.

I've also replaced the commit message with a more detailed version.
